Source/WTF/wtf/text/StringConcatenate.h

```diff
@@ -294,7 +294,13 @@ template<> class StringTypeAdapter<Unicode::CheckedUTF8, void> {
     unsigned length() const { return m_characters.lengthUTF16; }
     bool is8Bit() const { return m_characters.isAllASCII; }
     void writeTo(LChar* destination) const { memcpy(destination, m_characters.characters.data(), m_characters.lengthUTF16); }
+#ifndef __swift__
     void writeTo(UChar* destination) const { Unicode::convert(m_characters.characters, std::span { destination, m_characters.lengthUTF16 }); }
+#else
+    // FIXME: Only added for Swift C++ interop and UChar is always 16 bits so the bitwise_cast is fine.
+    // The real issue is that Swift considers char16_t as an alias for unsigned short, though char16_t is a type as of C++20. See rdar://136156228.
+    void writeTo(UChar* destination) const { Unicode::convert(m_characters.characters, std::span { bitwise_cast<char16_t *>(destination), m_characters.lengthUTF16 }); }
+#endif
```

Comment:
Do we need to call this from Swift? If not, probably better not to #define a Swift alternative.